cmd/backup/encrypt_archive.go

```diff
@@ -4,20 +4,87 @@
 package main
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path"
 
+	"github.com/ProtonMail/go-crypto/openpgp/armor"
 	openpgp "github.com/ProtonMail/go-crypto/openpgp/v2"
 	"github.com/offen/docker-volume-backup/internal/errwrap"
 )
 
-// encryptArchive encrypts the backup file using PGP and the configured passphrase.
-// In case no passphrase is given it returns early, leaving the backup file
+func readArmoredKeys(data []byte) (openpgp.EntityList, error) {
+	block, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	return block, nil
+}
+
+func (s *script) encryptAsymmetrically(outFile *os.File) (io.WriteCloser, func() error, error) {
+	if s.c.GpgPublicKeys == "" {
+		return nil, nil, nil
+	}
+
+	entityList, err := readArmoredKeys([]byte(s.c.GpgPublicKeys))
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, fmt.Sprintf("Error parsing key: %v", err))
+	}
+
+	armoredWriter, err := armor.Encode(outFile, "PGP MESSAGE", nil)
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, "error preparing encryption")
+	}
+
+	_, name := path.Split(s.file)
+	dst, err := openpgp.Encrypt(armoredWriter, entityList, nil, nil, &openpgp.FileHints{
+		FileName: name,
+	}, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return dst, func() error {
+		dst.Close()
+		armoredWriter.Close()
+		return nil
+	}, err
+}
+
+func (s *script) encryptSymmetrically(outFile *os.File) (io.WriteCloser, func() error, error) {
+	if s.c.GpgPassphrase == "" {
+		return nil, nil, nil
+	}
```

Comment:
As you now introduced checking for proper config in `encryptArchive`, I'm not sure this check is still needed. Is there still a way 'encryptSymmetrically` could be called _without_ a passphrase being set?

The same applies to the check in `encryptAsymmetrically`.

Reply:
yes, again orphaned code. I will remove it

Reply:
fixed in [96f996b](https://github.com/offen/docker-volume-backup/pull/456/commits/96f996b25947c66197177ee38823094b61fe98e2)